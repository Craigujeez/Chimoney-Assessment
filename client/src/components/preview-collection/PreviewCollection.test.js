import React from 'react';
import Enzyme,{ shallow } from 'enzyme';

import PreviewCollection from './PreviewCollection';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new ReactSixteenAdapter() });

describe('CollectionPreview component', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const mockRouteName = 'hats';

  beforeEach(() => {
    mockMatch = {
      path: '/shop'
    };

    mockHistory = {
      push: jest.fn()
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      routeName: mockRouteName,
      title: 'hats',
      items: []
    };

    wrapper = shallow(<PreviewCollection {...mockProps} />);
  });

  it('should render Preview Collection component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push with the right string when TitleContainer clicked', () => {
    wrapper.find('TitleContainer').simulate('click');

    expect(mockHistory.push).toHaveBeenCalledWith(
      `${mockMatch.path}/${mockRouteName}`
    );
  });
});