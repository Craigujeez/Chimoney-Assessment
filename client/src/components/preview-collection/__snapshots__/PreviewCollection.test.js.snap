// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component should render Preview Collection component 1`] = `undefined`;
